deploy/tekton/overlays/prod/task-patch.yaml

```diff
@@ -1,11 +1,7 @@
-apiVersion: tekton.dev/v1
-kind: Task
-metadata:
-  name: execute-ai-analysis
-spec:
-  params:
-    - name: CONTAINER_IMAGE
-      type: string
-      description: "Container image to use for SAST AI analysis"
-      default: "quay.io/ecosystem-appeng/sast-ai-workflow:latest"
-
+# JSON Patch to override CONTAINER_IMAGE default value for prod environment
+# Uses JSON Patch (RFC 6902) to replace only the default value without affecting other params
+# CONTAINER_IMAGE is at index 18 in the params array
+# This value is updated automatically in new releases
+- op: replace
+  path: /spec/params/18/default
```

Comment:
im afraid we are getting into trouble slowly with tthis indexes

i mean this can cause us by mistake a very serious hard to identify issue


if someone adds or modify parameters in some PR
the indexes get changed which means we will be needing to modify all the aptches using indexes to access


(i think this issue persist also in other overlays, but it just came to my mind)


i believe it will be great if we make an experiement and check if we can access parameters for replacement, by name instead by index.
maybe it does work